run_analysis.py

```diff
@@ -41,50 +35,87 @@ def makedirs(inpath):  # {{{
     return inpath  # }}}
 
 
+def check_generate(config, analysisName, mpasCore, analysisCategory=None):
+    # {{{
+    """
+    determine if a particular analysis of a particular core and (optionally)
+    category should be generated.
```

Comment:
It would be better to have a name / timestamp on this doc string if possible for consistency in other parts of the code.